cmd/object-api-common.go

```diff
@@ -313,6 +313,50 @@ func listObjectsNonSlash(ctx context.Context, bucket, prefix, marker, delimiter
 	return result, nil
 }
 
+func fsWalk(ctx context.Context, obj ObjectLayer, bucket, prefix string, doneCh chan struct{}, listDir ListDirFunc, getObjInfo func(context.Context, string, string) (ObjectInfo, error), getObjectInfoDirs ...func(context.Context, string, string) (ObjectInfo, error)) <-chan ObjectInfo {
+	walkResultCh := startTreeWalk(ctx, bucket, prefix, "", true, listDir, doneCh)
+
+	objInfoCh := make(chan ObjectInfo)
+	go func() {
+		defer close(objInfoCh)
+		for {
```

Comment:
Shouldn't there be a method to abort the walk when e.g. the client closes the connection - i.e. the given context is done?
Like:
```
for {
    select {
        case <-ctx.Done():
         default: // continue
    }
    walkResult, ok := <-walkResultCh
    if !ok {
      ...
}
```

Reply:
That's called doneCh which is already passed into startMergeWalks()

Reply:
We don't use context for such things anyways

Reply:
We do in few places, such as locking uses context.Done() but here there is an explicit doneCh to cancel the walker.

Reply:
It's already fixed